#! /bin/sh /usr/share/dpatch/dpatch-run
## 11-module-build.dpatch by  <meskes@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Build module correctly in linux-modules-extra

@DPATCH@

diff -ruN virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/Additions/linux/module/Makefile.module virtualbox-ose-1.5.2-dfsg2/src/VBox/Additions/linux/module/Makefile.module
--- virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/Additions/linux/module/Makefile.module	2007-10-18 13:07:28.000000000 +0200
+++ virtualbox-ose-1.5.2-dfsg2/src/VBox/Additions/linux/module/Makefile.module	2007-12-05 13:43:58.000000000 +0100
@@ -14,7 +14,7 @@
 #  be useful, but WITHOUT ANY WARRANTY of any kind.
 
 #
-MODULE = vboxadd
+override MODULE = vboxadd
 OBJS   = \
 	cmc.o \
 	hgcmcall.o \
@@ -92,6 +92,8 @@
  ifndef KBUILD_EXTMOD
   KBUILD_EXTMOD := $(shell pwd)
  endif
+# make sure we have a correct path
+ KBUILD_EXTMOD := $(subst $(MODULE),,$(KBUILD_EXTMOD))/$(MODULE)
  INCL    += $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
  export INCL
 endif
diff -ruN virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/Additions/linux/sharedfolders/Makefile.module virtualbox-ose-1.5.2-dfsg2/src/VBox/Additions/linux/sharedfolders/Makefile.module
--- virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/Additions/linux/sharedfolders/Makefile.module	2007-10-18 13:07:28.000000000 +0200
+++ virtualbox-ose-1.5.2-dfsg2/src/VBox/Additions/linux/sharedfolders/Makefile.module	2007-12-05 13:44:27.000000000 +0100
@@ -14,7 +14,7 @@
 #  be useful, but WITHOUT ANY WARRANTY of any kind.
 
 #
-MODULE = vboxvfs
+override MODULE = vboxvfs
 OBJS   = \
 	vfsmod.o \
 	vfs-utils.o \
@@ -99,6 +99,8 @@
  ifndef KBUILD_EXTMOD
   KBUILD_EXTMOD := $(shell pwd)
  endif
+# make sure we have a correct path
+ KBUILD_EXTMOD := $(subst $(MODULE),,$(KBUILD_EXTMOD))/$(MODULE)
  INCL    += $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
  export INCL
 endif
diff -ruN virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/HostDrivers/Support/linux/Makefile virtualbox-ose-1.5.2-dfsg2/src/VBox/HostDrivers/Support/linux/Makefile
--- virtualbox-ose-1.5.2-dfsg2.orig/src/VBox/HostDrivers/Support/linux/Makefile	2007-10-18 13:07:28.000000000 +0200
+++ virtualbox-ose-1.5.2-dfsg2/src/VBox/HostDrivers/Support/linux/Makefile	2007-12-05 13:43:14.000000000 +0100
@@ -40,7 +40,7 @@
 endif
 
 
-MODULE = vboxdrv
+override MODULE = vboxdrv
 OBJS   = \
 	linux/SUPDrv-linux.o \
 	SUPDRVShared.o \
